code/hud/hudlock.cpp

```diff
@@ -277,7 +277,7 @@ void hud_lock_reset(float lock_time_scale)
     
 	if ((swp->current_secondary_bank >= 0) && (swp->secondary_bank_weapons[swp->current_secondary_bank] >= 0)) {
 		Assert(swp->current_secondary_bank < MAX_SHIP_SECONDARY_BANKS);
-		Assert(swp->secondary_bank_weapons[swp->current_secondary_bank] < MAX_WEAPON_TYPES);
+		Assert(swp->secondary_bank_weapons[swp->current_secondary_bank] < static_cast<int>(Weapon_info.size()));
```

Comment:
I don't like having this `static_cast<int>(Weapon_info.size())` used all over the code since it adds a lot of clutter. Maybe this could be replaced with a single function that is then called where appropriate?

Reply:
I don't like it either, but we already have `static_cast<int>(Ship_info.size())` all over the code.  That's why I did this.  Of course, we could replace *both* uses...

Reply:
We are going to need that more and more should we ever make the vector truly dynamic so I think it would be good to have a single function that returns the size as an int. Since this is about the weapons I would see doing the same for ships out of scope of this PR.

Reply:
Why not approve this PR the way it is, and then I'll make a new PR specifically for removing the static_cast on both ships and weapons.

Reply:
Works for me.